src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -411,28 +368,27 @@ def get_observation(self, event: Event) -> pd.DataFrame:
         )
         simplified_race_ethnicity[simplified_race_ethnicity.isnull()] = "Other"
 
+        # pr is 1.0 for infants with mother on WIC
         child_covered_pr = (
-            pop_u1.guardian_1.isin(pop_included_mothers.index)
-            | pop_u1.guardian_2.isin(pop_included_mothers.index)
-        ).astype(
-            float
-        )  # pr is 1.0 for infants with mother on WIC
+            pop_u1["guardian_1"].isin(pop_included_mothers["simulant_id"])
+            | pop_u1["guardian_2"].isin(pop_included_mothers["simulant_id"])
+        ).astype(float)
         for race_ethnicity in self.WIC_RACE_ETHNICITIES:
             race_eth_rows = simplified_race_ethnicity == race_ethnicity
-
-            N = np.sum(race_eth_rows)  # total number of infants in this race group
-            k = np.sum(
-                race_eth_rows & (child_covered_pr == 1)
-            )  # number included because their mother is on WIC
-            if k < N:
+            num_infants_total = np.sum(
+                race_eth_rows
+            )  # total number of infants in this race group
+            # number included because their mother is on WIC
+            num_infants_included = np.sum(race_eth_rows & (child_covered_pr == 1))
+            if num_infants_included < num_infants_total:
                 pr_covered = data_values.COVERAGE_PROBABILITY_WIC[0]
+                # keep pr of 1.0 for the num_infants_included infants with mother
+                # on WIC and rescale probability for the remaining individuals
+                # so that expected number of infants on WIC matches target
                 child_covered_pr[race_eth_rows] = np.maximum(
-                    child_covered_pr[
-                        race_eth_rows
-                    ],  # keep pr of 1.0 for the k infants with mother on WIC
-                    (pr_covered[race_ethnicity] * N - k)
-                    / (N - k)  # rescale probability for the remaining individuals
-                    # so that expected number of infants on WIC matches target
+                    child_covered_pr[race_eth_rows],
+                    (pr_covered[race_ethnicity] * num_infants_total - num_infants_included)
+                    / (num_infants_total - num_infants_included),
                 )
         pop_included[0] = self.randomness.filter_for_probability(pop_u1, child_covered_pr)
```

Comment:
pop_included is a dict and should always be the same order but will the groupby object also always be ordered the same?

Reply:
I'm not sure I understand the question here. pop_included is a dict where keys are the ages. This line is just adding the values for age 0.